python/ray/serve/backend_state.py

```diff
@@ -96,45 +97,56 @@ def __set_state__(self, d: Dict[Any, Any]) -> None:
 
     @property
     def replica_tag(self) -> str:
-        return str(self._replica_tag)
+        return self._replica_tag
 
     @property
     def backend_tag(self) -> str:
         return self._backend_tag
 
     @property
-    def actor_handle(self) -> ActorHandle:
-        return ray.get_actor(
-            self._actor_name, namespace=self._controller_namespace)
+    def actor_handle(self) -> Optional[ActorHandle]:
+        if not self._actor_handle:
+            try:
+                self._actor_handle = ray.get_actor(
+                    self._actor_name, namespace=self._controller_namespace)
+            except ValueError:
+                self._actor_handle = None
 
-    def _handle_placement_group(self, backend_info):
-        self._actor_resources = backend_info.replica_config.resource_dict
+        return self._actor_handle
 
-        # Feature flagging because of placement groups doesn't handle
-        # newly added nodes.
-        # https://github.com/ray-project/ray/issues/15801
-        if USE_PLACEMENT_GROUP:
+    def create_placement_group(self, placement_group_name: str,
+                               actor_resources: dict) -> PlacementGroup:
+        logger.debug("Creating placement group '{}' for deployment '{}'".
+                     format(placement_group_name, self.backend_tag) +
+                     f" component=serve deployment={self.backend_tag}")
+        placement_group = ray.util.placement_group(
+            [actor_resources],
+            lifetime="detached" if self._detached else None,
+            name=placement_group_name)
+
+        return placement_group
+
+    def get_placement_group(self,
+                            placement_group_name) -> Optional[PlacementGroup]:
+        if not self._placement_group:
             try:
-                self._placement_group = ray.util.get_placement_group(
-                    self._placement_group_name)
+                placement_group = ray.util.get_placement_group(
+                    placement_group_name)
```

Comment:
why don't we set `self._placement_group` in this method? does it get set elsewhere?

Reply:
this method smells kind of funny to me, why are we passing in the placement group name if we don't use it in the case where it's already set in `self._placement_group`? shouldn't it just be a field in the class? any reason why this doesn't exactly mirror `self.actor_handle`?

Reply:
I'm trying to split creating / fetching calls so that they're not mixed with instance variable assignment. Current implementation of actor handle is making a GCS call; but also isn't clear which method is idempotent and where re-assignment happens exactly, such as a call to `get_placement_group` with new / wrong name might lead to creating a new instance and re-assign to backend state instance implicitly.

To be consistent i changed both actor handle and placement group to be local variable first; Then all helper functions either get or create both re-assign instance variable and will return cached results if one already exists.

```
@property
def actor_handle(self) -> ActorHandle:
    return ray.get_actor(self._actor_name)
```